src/app/sidebar/Sidebar.tsx

```diff
@@ -1,63 +1,60 @@
-import { Flex, Heading } from '@chakra-ui/react';
-import React from 'react';
-import InfiniteScroll from 'react-infinite-scroll-component';
+import { Box, Flex } from '@chakra-ui/react';
+import { useCallback } from 'react';
 import { Card } from './components/Card';
 import { CardDropDown } from './components/CardDropDown';
-import { Course, Term, useGetCourses, useSubjects } from '../../fetchers'
-
-// import StyledHeader from './Header.styles';
-
+import { Course, Term, useGetCourses, useSubjects } from '../../fetchers';
 
 export interface SidebarProps {
   /**
    * Term Selected
    * Determines what term the subjects and courses are from
    */
   term: Term;
+  setPid?: (pid: string) => void;
+  pid: string;
 }
 
-export function Sidebar({ term }: SidebarProps): JSX.Element {
+export function Sidebar({ term, pid, setPid }: SidebarProps): JSX.Element {
+  const { data: subjects } = useSubjects({ term: term });
+  const { data: courses } = useGetCourses({ term: term });
 
-  const { data: subjects, loading: loadingSubjects, error: errorSubjects } = useSubjects({ term: term });
-  const { data: courses, loading: loadingCourses, error: errorCourses } = useGetCourses({ term: term });
+  const parsedCourses =
+    courses?.reduce((dict, course) => {
+      const subject = course.subject;
+      if (!(subject in dict)) {
+        dict[subject] = [];
+      }
+      dict[subject].push(course);
+      return dict;
+    }, {} as { [subject: string]: Course[] }) ?? {};
 
-  const parsedCourses = courses?.reduce((dict, course) => {
-    const subject = course.subject;
-    if (!(subject in dict)) {
-      dict[subject] = [];
-    }
-    dict[subject].push(course);
-    return dict;
-  }, {} as { [subject: string]: Course[] }) ?? {};
+  const handleClick = useCallback(
```

Comment:
im not sure how this works? mind explaining it please

Reply:
This is the function that is called when a course card is clicked. The function `setPid` changes the `pid` state to the selected courses corresponding `pid`, which is then used in `Content.tsx` to fetch the course information. [`useCallback`](https://reactjs.org/docs/hooks-reference.html#usecallback) is a react hook which is typically used when you're changing a state that's used in different components.

Reply:
Not sure if it helps make a dent in the performance though in this case.